pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -773,48 +821,125 @@ func (a *AmbientIndex) handlePods(pods []*v1.Pod, c *Controller) {
 	}
 }
 
-func (a *AmbientIndex) handleService(obj any, isDelete bool, c *Controller) map[model.ConfigKey]struct{} {
+func (a *AmbientIndex) handleService(obj any, isDelete bool, c *Controller) sets.Set[model.ConfigKey] {
 	svc := controllers.Extract[*v1.Service](obj)
+	updates := sets.New[model.ConfigKey]()
+
+	if svc.Labels[constants.ManagedGatewayLabel] == constants.ManagedGatewayMeshControllerLabel {
+		scope := model.WaypointScope{Namespace: svc.Namespace, ServiceAccount: svc.Annotations[constants.WaypointServiceAccount]}
+
+		// TODO get IP+Port from the Gateway CRD
+		// https://github.com/istio/istio/issues/44230
+		if svc.Spec.ClusterIP == "None" {
+			// TODO handle headless Service
+			return updates
+		}
+		svcIP := netip.MustParseAddr(svc.Spec.ClusterIP)
+		addr := &workloadapi.GatewayAddress{
+			Destination: &workloadapi.GatewayAddress_Address{
+				Address: &workloadapi.NetworkAddress{
+					Network: c.Network(svcIP.String(), make(labels.Instance, 0)).String(),
+					Address: svcIP.AsSlice(),
+				},
+			},
+			Port: uint32(svc.Spec.Ports[0].Port),
```

Comment:
do we have a reserved port name for HBONE traffic we should be searching for instead of just taking the first one?

Reply:
the 15008 is the default - this is part of the TODO surrounding getting the port and address for the waypoint from the Gateway object

Reply:
cc @howardjohn do we need a more robust solution in place until https://github.com/kubernetes-sigs/gateway-api/pull/1653 lands?  Is `Ports[0]` enough?  hardcode to 15008? check specifically for `https-hbone` port name?

Reply:
I'm thinking hardcode to 15008 and then take the first port named `hbone`, if any is there. but im flexible. any special "take first port or port named hbone" behavior might put us in a weird place for backwards compat, so in that scenario i might even push for just taking 15008 hard-coded for now if https://github.com/kubernetes-sigs/gateway-api/pull/1653 will land soon